universal-application-tool-0.0.1/app/views/FileUploadViewStrategy.java

```diff
@@ -60,6 +70,102 @@ protected Tag renderQuestion(
     return applicantQuestionRendererFactory.getRenderer(question).render(params);
   }
 
+  /**
+   * Renders a form submit button for continue form if an uploaded file is present.
+   *
+   * <p>See {@link renderDeleteAndContinueFileUploadForms}.
+   */
+  Optional<Tag> maybeRenderContinueButton(Params params) {
+    if (!hasUploadedFile(params)) {
+      return Optional.empty();
+    }
+    Tag button =
+        submitButton(params.messages().at(MessageKey.BUTTON_KEEP_FILE.getKeyName()))
+            .attr(FORM, FILEUPLOAD_CONTINUE_FORM_ID)
+            .withClasses(ApplicantStyles.BUTTON_BLOCK_NEXT)
+            .withId(FILEUPLOAD_CONTINUE_BUTTON_ID);
+    return Optional.of(button);
+  }
+
+  /**
+   * Renders a form submit button for delete form if the file upload question is optional.
+   *
+   * <p>If an uploaded file is present, render the button text as delete. Otherwise, skip.
+   *
+   * <p>See {@link renderDeleteAndContinueFileUploadForms}.
+   */
+  Optional<Tag> maybeRenderSkipOrDeleteButton(Params params) {
+    if (hasAtLeastOneRequiredQuestion(params)) {
+      // If the file question is required, skip or delete is not allowed.
+      return Optional.empty();
+    }
+    String buttonText = params.messages().at(MessageKey.BUTTON_SKIP_FILEUPLOAD.getKeyName());
+    String buttonId = FILEUPLOAD_SKIP_BUTTON_ID;
+    if (hasUploadedFile(params)) {
+      buttonText = params.messages().at(MessageKey.BUTTON_DELETE_FILE.getKeyName());
+      buttonId = FILEUPLOAD_DELETE_BUTTON_ID;
+    }
+    Tag button =
+        submitButton(buttonText)
+            .attr(FORM, FILEUPLOAD_DELETE_FORM_ID)
+            .withClasses(ApplicantStyles.BUTTON_REVIEW)
+            .withId(buttonId);
+    return Optional.of(button);
+  }
+
+  /**
+   * Returns two hidden forms for navigating through a file upload block without uploading a file.
+   *
+   * <p>Delete form sends an update with an empty file key. An empty file key erases the existing
+   * file key if one is present. In either case, the file upload question is marked as seen but
+   * unanswered, namely skipping the file upload. This is only allowed for an optional question.
+   *
+   * <p>Continue form sends an update with the currently stored file key, the same behavior as an
+   * applicant re-submits a form without changing their answer. Continue form is only used when an
+   * existing file (and file key) is present.
+   */
+  Tag renderDeleteAndContinueFileUploadForms(Params params) {
```

Comment:
suggestion (blocking): for all methods in this class that don't have an access modifier, verify they cannot have a more restrictive level access (likely `protected`)

Reply:
Restricted them all to private